Auth/app/src/main/java/com/example/Auth/BioActivity.kt

```diff
@@ -0,0 +1,316 @@
+package com.example.Auth
+
+
+import android.app.Activity
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.os.AsyncTask
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.Auth.databinding.ActivityBioBinding
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import okhttp3.*
+import java.io.ByteArrayOutputStream
+import java.util.*
+
+private val jsonFormat = Json {
+    coerceInputValues = true; ignoreUnknownKeys = true
+}
+
+class BioActivity : AppCompatActivity(), OnKnopkaClickListener {
+    lateinit var binding: ActivityBioBinding
+    private val adapter = KnopkaFeedAdapter(this)
+//    private var ind: Int = 0;
+
+    class MainActivityUnits {
+
+        var changeInfoButton: Button? = null
+        var textViewName: TextView? = null
+        var textViewBio: TextView? = null
+        var imageViewProfilePic: ImageView? = null
+        var googleLogOutButton: Button? = null
+        var profilePicBitMap: Bitmap? = null
+        var token: String? = null
+        var id: Int = 1
+    }
+
+    private var units: MainActivityUnits = MainActivityUnits()
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun bitMapToBase64String(bitMap: Bitmap?): String? {
+        val byteArrayOutputStream = ByteArrayOutputStream()
+        bitMap?.compress(Bitmap.CompressFormat.JPEG, 10, byteArrayOutputStream)
+        // TODO: what quality to use?
+        val imageBytes: ByteArray = byteArrayOutputStream.toByteArray()
+        return Base64.getEncoder().encodeToString(imageBytes)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun base64StringToBitMap(str: String?): Bitmap? {
+        val imageBytes = Base64.getDecoder().decode(str)
+        return BitmapFactory.decodeByteArray(imageBytes, 0, imageBytes.size)
+    }
+
+    fun stringToButtons(knopkasString: String): List<Knopka> {
+        val knopkaIdsList =
+            knopkasString.removeSurrounding("[", "]")
```

Comment:
JSON should never be processed as a string. You can parse it directly to list, no need to manually split into separate rows.